src/peer_select.cc

```diff
@@ -256,154 +390,131 @@ PeerSelector::CheckAlwaysDirectDone(allow_t answer, void *data)
 bool
 PeerSelector::selectionAborted()
 {
-    if (interestedInitiator())
+    if (callback_ && !callback_->canceled())
         return false;
 
     debugs(44, 3, "Aborting peer selection: Initiator gone or lost interest.");
-    delete this;
     return true;
 }
 
-/// A single DNS resolution loop iteration: Converts selected FwdServer to IPs.
-void
-PeerSelector::resolveSelected()
+bool
+PeerSelector::accessCheckCached(CachePeer *p, allow_t &answer)
```

Comment:
Please use `const CachePeer *`, if possible.

Reply:
ok

Reply:
Fixed. Also please check whether the method `PeerSelector::accessCheckCached()` itself could be converted into constant.

Reply:
I made it const.